### DISCUSSION TITLE
topic count limitation

### DISCUSSION THREAD

**Discussion by shaojun** [emqx/emqx#8261] [General]
Assume I have a client application keep publishing data on a new topic(like append a GUID on topic) **every time**, What is the impact for this behavior, and is there such a `topic max count` limit in broker?

**Comment by lafirest** [comment]
There has no soft limit for `max topic count`, the limit depends on your memory and Erlang VM settings.
But I think maybe you shouldn't do this, topics should suit your business, and too many topics will make the route slow down

**Reply by shaojun** [reply]
does emqx support like 'auto clean unused topics'?

**Reply by lafirest** [reply]
In EMQX there is has no `unused topics`, this is an MQTT broker, and only subscribed topics will manage by EMQX router

**Reply by shaojun** [reply]
so even the topics which with long time no data published in will forever stayed in broke's memory, i believe always to call the unsub() from subscriber will keep the efficiency, correct?

**Reply by lafirest** [reply]
No, it's not like that, it only depends on the subscription not publish. You can publish everywhere, if the topic you published to does not exist, this message will be dropped

**Reply by lafirest** [reply]
And most of the time, you don't need to care about the memory consumption of the topics

**Comment by zmstone** [comment]
Messages published to non-existing topics (no subscribers) will be black-hold by EMQX.

**Reply by zmstone** [reply]
or depending on the flags, a negative ack will be sent back to the client.

**Comment by ismaelhamed** [comment]
This seems like a reasonable use case for IoT scenarios, where you might have a topic (queue) per device. A soft limit in these topics would prevent any actor from queueing up too many messages for a particular device (broker could reject new messages).
